python-checks/src/main/java/org/sonar/python/checks/ExceptGroupShouldNotCaughtExceptionGroupCheck.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.tree.ExceptClause;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.Tuple;
+
+import static org.sonar.plugins.python.api.tree.Tree.Kind.EXCEPT_GROUP_CLAUSE;
+
+@Rule(key = "S6468")
+public class ExceptGroupShouldNotCaughtExceptionGroupCheck extends PythonSubscriptionCheck {
+  private static final Set<String> EXCEPTION_GROUP = Set.of("ExceptionGroup", "BaseExceptionGroup");
+  private static final String MESSAGE = "Exception group cannot be caught by an except*.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(EXCEPT_GROUP_CLAUSE, ctx -> {
+      ExceptClause exceptClause = (ExceptClause) ctx.syntaxNode();
+
+      if (hasAnyExceptionGroup(exceptClause.exception())) {
+        ctx.addIssue(exceptClause.exceptKeyword(), MESSAGE);
+      }
+    });
+  }
+
+  public boolean hasAnyExceptionGroup(Expression exception) {
+    if (isExceptionGroup(exception)) {
+      return true;
+    }
+    if (exception.is(Tree.Kind.TUPLE)) {
+      Tuple exceptionTuple = (Tuple) exception;
+      for (Expression exceptionEl : exceptionTuple.elements()) {
+        if (isExceptionGroup(exceptionEl)) {
+          return true;
+        }
+      }
+      return false;
+    }
+    return false;
+  }
+
+  public boolean isExceptionGroup(Expression exception) {
+    return exception.is(Tree.Kind.NAME) && EXCEPTION_GROUP.contains(((Name) exception).name());
```

Comment:
I think this is fine for the time being, however let's put a "TODO" to change this logic to use type or symbol once we will have it from typeshed.

Also, can you please document the FP we might have when overriding symbol with a unit test?

Reply:
You mean once we will have it from the typeshed version we are going to upgrade in this sprint ?
I will add the TODO + create a ticket to have this fixed.